rxjava/src/main/java/com/elpassion/android/commons/rxjava/ScheduledRxJavaCallAdapterFactory.kt

```diff
@@ -0,0 +1,28 @@
+package com.elpassion.android.commons.rxjava
+
+import retrofit2.Call
+import retrofit2.CallAdapter
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory
+import rx.Observable
+import rx.Scheduler
+import java.lang.reflect.Type
+
+class ScheduledRxJavaCallAdapterFactory(private val observeOn: Scheduler,
+                                        private val subscribeOn: Scheduler) : CallAdapter.Factory() {
+
+    private val rxJavaCallAdapter by lazy { RxJavaCallAdapterFactory.createWithScheduler(subscribeOn) }
+
+    override fun get(returnType: Type, annotations: Array<out Annotation>, retrofit: Retrofit): CallAdapter<*>? {
+        if (getRawType(returnType) != Observable::class.java) {
+            return null
+        }
+        @Suppress("UNCHECKED_CAST")
+        val delegate = rxJavaCallAdapter.get(returnType, annotations, retrofit) as CallAdapter<Observable<*>>
```

Comment:
Yes but if anyone will not pass the rxJavaCallAdapter this adapter will crash, won't it?